File: healthtrack/test_admin_login_fix.js
const http = require('http');

// Test admin login with correct credentials to verify our fix
console.log('Testing admin login with username "edwin" and password "admin"...');

const options = {
  hostname: '10.243.17.91',
  port: 3000,
  path: '/admin/login',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json'
  }
};

const req = http.request(options, (res) => {
  console.log(`Status Code: ${res.statusCode}`);
  console.log(`Content-Type: ${res.headers['content-type']}`);
  
  let data = '';
  res.on('data', (chunk) => {
    data += chunk;
  });
  
  res.on('end', () => {
    console.log(`Response Body: ${data}`);
    
    // Try to parse JSON
    try {
      const jsonData = JSON.parse(data);
      console.log('✅ Successfully parsed JSON response:');
      console.log(JSON.stringify(jsonData, null, 2));
    } catch (parseError) {
      console.log('❌ JSON Parse Error:', parseError.message);
    }
  });
});

req.on('error', (e) => {
  console.error(`Problem with request: ${e.message}`);
});

// Send correct login credentials
req.write(JSON.stringify({
  username: 'edwin',
  password: 'admin'
}));
req.end();

File: healthtrack/test_admin_login_fix.dart
import 'dart:convert';
import 'package:http/http.dart' as http;

void main() async {
  // Test the admin login with proper error handling
  final String baseUrl = 'http://10.243.17.91:3000';
  final String loginEndpoint = '/admin/login';
  
  print('Testing admin login with username "edwin" and password "admin"...');
  
  try {
    final response = await http.post(
      Uri.parse('$baseUrl$loginEndpoint'),
      headers: {'Content-Type': 'application/json'},
      body: json.encode({
        'username': 'edwin',
        'password': 'admin',
      }),
    ).timeout(Duration(seconds: 10));
    
    print('Status Code: ${response.statusCode}');
    print('Content-Type: ${response.headers['content-type']}');
    print('Response Body: ${response.body}');
    
    // Test our JSON parsing fix
    if (response.body.isEmpty) {
      print('❌ Server returned an empty response');
      return;
    }
    
    if (!response.headers.containsKey('content-type') || 
        !response.headers['content-type']!.contains('application/json')) {
      print('❌ Server returned a non-JSON response');
      return;
    }
    
    try {
      final data = json.decode(response.body);
      print('✅ Successfully parsed JSON response:');
      print('Response data: $data');
      
      // Check if login was successful
      bool isSuccess = false;
      if (data['success'] is bool) {
        isSuccess = data['success'];
      } else if (data['success'] is String) {
        isSuccess = data['success'].toLowerCase() == 'true';
      } else if (data['success'] is int) {
        isSuccess = data['success'] == 1;
      }
      
      if (isSuccess) {
        print('✅ Admin login successful!');
      } else {
        print('❌ Admin login failed: ${data['message'] ?? 'Unknown error'}');
      }
    } catch (parseError) {
      print('❌ JSON Parse Error: $parseError');
      print('Response body: ${response.body}');
    }
  } catch (e) {
    print('Connection error: $e');
  }
}